test/e2e/csv_e2e_test.go

```diff
@@ -2664,7 +2664,136 @@ var _ = Describe("CSV", func() {
+					Plural:   crdPlural,
+					Singular: crdPlural,
+					Kind:     crdPlural,
+					ListKind: "list" + crdPlural,
+				},
+				Scope: "Namespaced",
+			},
+		})
+		defer cleanupCRD()
+		require.NoError(GinkgoT(), err)
+
+		// Create "current" CSV
+		nginxName := genName("nginx-")
+		strategy := v1alpha1.StrategyDetailsDeployment{
+			DeploymentSpecs: []v1alpha1.StrategyDeploymentSpec{
+				{
+					Name: genName("dep-"),
+					Spec: newNginxDeployment(nginxName),
+				},
+			},
+		}
+
+		csv := v1alpha1.ClusterServiceVersion{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       v1alpha1.ClusterServiceVersionKind,
+				APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: genName("csv"),
+			},
+			Spec: v1alpha1.ClusterServiceVersionSpec{
+				MinKubeVersion: "0.0.0",
+				InstallModes: []v1alpha1.InstallMode{
+					{
+						Type:      v1alpha1.InstallModeTypeOwnNamespace,
+						Supported: true,
+					},
+					{
+						Type:      v1alpha1.InstallModeTypeSingleNamespace,
+						Supported: true,
+					},
+					{
+						Type:      v1alpha1.InstallModeTypeMultiNamespace,
+						Supported: true,
+					},
+					{
+						Type:      v1alpha1.InstallModeTypeAllNamespaces,
+						Supported: true,
+					},
+				},
+				InstallStrategy: v1alpha1.NamedInstallStrategy{
+					StrategyName: v1alpha1.InstallStrategyNameDeployment,
+					StrategySpec: strategy,
+				},
+				CustomResourceDefinitions: v1alpha1.CustomResourceDefinitions{
+					Owned: []v1alpha1.CRDDescription{
+						{
+							Name:        crdName,
+							Version:     "v1alpha1",
+							Kind:        crdPlural,
+							DisplayName: crdName,
+							Description: "In the cluster",
+						},
+					},
+				},
+			},
+		}
+
+		cleanupCSV, err := createCSV(c, crc, csv, testNamespace, true, false)
+		require.NoError(GinkgoT(), err)
+		defer cleanupCSV()
+
+		// Wait for current CSV to succeed
+		_, err = fetchCSV(crc, csv.Name, testNamespace, csvSucceededChecker)
+		require.NoError(GinkgoT(), err)
+
+		// Should have created deployment
+		dep, err := c.GetDeployment(testNamespace, strategy.DeploymentSpecs[0].Name)
+		require.NoError(GinkgoT(), err)
+		require.NotNil(GinkgoT(), dep)
+
+		// Create "updated" CSV
+		strategyNew := v1alpha1.StrategyDetailsDeployment{
+			DeploymentSpecs: []v1alpha1.StrategyDeploymentSpec{
+				{
+					// Same name
+					Name: strategy.DeploymentSpecs[0].Name,
+					// Different spec
+					Spec: newNginxDeployment(nginxName),
+				},
+			},
+		}
+
+		// Fetch the current csv
+		fetchedCSV, err := fetchCSV(crc, csv.Name, testNamespace, csvSucceededChecker)
+		require.NoError(GinkgoT(), err)
+
+		// Update csv with modified deployment spec
+		fetchedCSV.Spec.InstallStrategy.StrategySpec = strategyNew
+
+		// Update the current csv
+		_, err = crc.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Update(context.TODO(), fetchedCSV, metav1.UpdateOptions{})
+		require.NoError(GinkgoT(), err)
```

Comment:
Could this all be done in an Eventually() command? As it is, the update command may occasionally fail if another resource modifies the CSV before it is fetched.

Reply:
sorry I missed this msg, I have update to include eventually() block https://github.com/operator-framework/operator-lifecycle-manager/pull/1817/files#diff-e94bc9e5ebd2b6db20f851288184630432208a5a772cf0515935707ad3660799R3034